libs/parallelism/algorithms/tests/regressions/for_each_datapar.cpp

```diff
@@ -0,0 +1,54 @@
+#include <hpx/algorithm.hpp>
```

Comment:
Does this test something that the existing tests didn't test (after all the existing tests segfaulted)? If not we can leave this out. If yes is would this test be useful to add to all the other `for_each` tests instead? And if yes, but the file will stay here it needs a license header.

Reply:
the existing tests starts vectorisation at only one iterator i.e begin(), and gets aligned with prologue loop, but this has very less chances to check segfaults, checking at more iterator positions assures that iterator at different positions can be adjusted for alignment and then go for vectorisation.

Reply:
@msimberg Would you recommend adding this new test, or update the existing unit tests with these changes?. Now all the seg faults are fixed.

Reply:
I see, thanks for the explanation. Let's keep this as it is then! It does still need the license header though.

Reply:
@msimberg Done!